### PR TITLE
Allow use first IP of ipv4 range if anygw not enabled

### PR DESCRIPTION
**Pull request by p4u**
IP like 10.0.0.1/24 should be possible as main address if anygw not installed nor enabled.

Signed-off-by: p4u <p4u@dabax.net>

### CODE REVIEW

File: packages/lime-system/files/usr/lib/lua/lime/network.lua [hunk 1]
```diff
@@ -80,8 +80,15 @@ function network.primary_address(offset)
...
 	--! Generated address is network address like 192.0.2.0/24 ?
 	local invalid = ipv4_template:equal(mc:network()) and "NETWORK"
-	--! Generated address is the one reserved for anygw like 192.0.2.1/24 ?
-	invalid = invalid or ipv4_template:equal(mc:minhost()) and "ANYGW"
+	--! If anygw enabled, generated address is the one reserved for anygw like 192.0.2.1/24 ?
+	if utils.isModuleAvailable("lime.proto.anygw") then
+		local generalProtocols = config.get("network", "protocols")
+		for _,protocol in pairs(generalProtocols) do
+			if protocol == 'anygw' then
+				invalid = invalid or ipv4_template:equal(mc:minhost()) and "ANYGW"
```
Comment (G10h4ck) [inline packages/lime-system/files/usr/lib/lua/lime/network.lua]: At first glance I would say a `break` should be added just after this line ;)

Comment (p4u) [reply packages/lime-system/files/usr/lib/lua/lime/network.lua]: Upss, you'r right. Fixed

**Comment by p4u** [comment]
ping @G10h4ck

**Comment by G10h4ck** [comment]
Looking at it again I realized should't it be the same for IPv6 too ?

**Comment by p4u** [comment]
Probably. But this PR refers specifically to the IPv4 range. Another PR for the IPv6 might be sent.

@G10h4ck merging is blocked until you mark the requested changes as solved (which IMO they are).

**Comment by ilario** [comment]
The check for invalid IPv6 has not been implemented yet, as the probability of a collision is less likely.

**Pull request merged**